tsconfig.all.jsonc

```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "allowUnreachableCode": false, // Prevents accidental execution of unreachable code.
+    "allowUnusedLabels": false, // Disallows unused labels to avoid confusion in code.
+    "noFallthroughCasesInSwitch": true, // Ensures every switch case explicitly handles its execution flow.
+    "noUnusedLocals": true, // Reports unused local variables to keep the code clean.
+    "noErrorTruncation": true, // Prevents truncation of error messages for better debugging.
+    "isolatedModules": true, // Ensures each module can be transpiled independently for better tooling support.
+    "verbatimModuleSyntax": true, // Enforces explicit use of `type` imports for type-only dependencies.
+    "module": "nodenext", // Uses the ES module system compatible with Node.js.
+    "moduleResolution": "nodenext", // Resolves modules as per Node.js ESM standards.
+    "resolveJsonModule": true, // Allows importing JSON files as modules.
+    "types": ["vite/client", "vitest/globals", "node", "vitest"], // Provides Node.js global types and includes Vitest types.
+    "jsx": "react", // Specifies JSX transformation method for React applications.
+    "noEmitOnError": true, // Prevents output generation if there are TypeScript errors.
+    "incremental": true, // Enables incremental builds for better performance.
+    "skipLibCheck": true, // Skips type checking of library declaration files to speed up compilation.
+    "allowJs": true, // Allows TypeScript to compile JavaScript files.
+    "checkJs": false, // Disables type checking on .js files.
+    "noImplicitReturns": true, // Ensures all functions return a value (e.g., prevents missing return in `if` branches).
```

Comment:
This one we can skip. We've decided not to do it in spark but we haven't removed the setting yet.